### PR TITLE
Feature/box range

### PR DESCRIPTION
**Pull request by vlach1989**

### CODE REVIEW

File: src/ReactLeafletMap/index.js [hunk 1]
```diff
@@ -70,86 +87,113 @@ class ReactLeafletMap extends React.PureComponent {
...
     }
 
-    componentDidMount() {
-        // Hack for ugly 1px tile borders in Chrome
-        // Version of Leaflet package in dependencies should match version used by react-leaflet
-        let originalInitTile = L.GridLayer.prototype._initTile;
-        L.GridLayer.include({
-            _initTile: function (tile) {
-                originalInitTile.call(this, tile);
+    setZoomLevelsBounds(width, height) {
```
Comment (tmdmc) [inline src/ReactLeafletMap/index.js:90]: Should this be called on props change?

File: src/ReactLeafletMap/index.js [hunk 2]
```diff
@@ -70,86 +87,113 @@ class ReactLeafletMap extends React.PureComponent {
...
+        return (
+            <>
+                <ReactResizeDetector
+                    handleHeight
+                    handleWidth
+                    onResize={this.onResize}
+                    refreshMode="debounce"
+                    refreshRate={500}
+                />
+                {this.state.width && this.state.height ? this.renderMap() : null}
```
Comment (tmdmc) [inline src/ReactLeafletMap/index.js:164]: Will this work with SSR?

File: src/WorldWindMap/navigator/helpers.js [hunk 3]
```diff
@@ -1,8 +1,9 @@
+import viewUtils from "../../utils/view";
 
 function getChangedViewParams(prev, next) {
 		let changed = {};
 
-		if (prev.boxRange !== next.boxRange){
+		if (prev.boxRange !== next.boxRange && Math.abs(prev.boxRange - next.boxRange) > 1){
```
Comment (tmdmc) [inline src/WorldWindMap/navigator/helpers.js:6]:
```suggestion
		// check for boxRange change, disregard if change is too small
		if (prev.boxRange !== next.boxRange && Math.abs(prev.boxRange - next.boxRange) > 1){
```

File: src/constants/view.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
+const averageLatitude = 50;
```
Comment (tmdmc) [inline src/constants/view.js:1]: document

File: src/ReactLeafletMap/index.js (hunk 1 above)
Comment (vlach1989) [reply src/ReactLeafletMap/index.js:90]: It should. However, in react-leaflet has the Map component so-called dynamic and static props and minZoom and maxZoom are for some reason static. Therefore this is called on initial resize only.

File: src/ReactLeafletMap/index.js (hunk 2 above)
Comment (vlach1989) [reply src/ReactLeafletMap/index.js:164]: No idea. We should test this before production version release.

**Pull request merged**